os/binfmt/libelf/libelf_symbols.c

```diff
@@ -203,6 +203,32 @@ int elf_findsymtab(FAR struct elf_loadinfo_s *loadinfo)
 	return OK;
 }
 
+/****************************************************************************
+ * Name: elf_readsymtab
+ *
+ * Description:
+ *   Read the ELF symbol table into memory.
+ *
+ * Input Parameters:
+ *   loadinfo - Load state information
+ *
+ ****************************************************************************/
+void elf_readsymtab(FAR struct elf_loadinfo_s *loadinfo)
```

Comment:
This is almost same as elf_readreltab. How about making common function for avoiding duplications? I mean not for whole code, just duplicated code.

Reply:
I dont think its a good idea to combine the two functions. Even if we combine them, most of the lines in the function need to be repeated according to the data structure to be used. For example, elf_readsym() and elf_readrel() are performing similar operation, but still they are maintained as separate API's.

Reply:
Better to keep two separate function